src/cpp/src/gguf_utils/building_blocks.cpp

```diff
@@ -255,12 +255,84 @@ std::pair<Output<ov::Node>, Output<ov::Node>> rope_emb(
     };
 }
 
-//std::tuple<Output<Node>, std::vector<std::shared_ptr<VariableState>>, std::pair<Output<Node>, Output<Node>>, Output<Node>>
+// Helper function to split heads
+// There are q_norm k_norm in Qwen3, if key_name + ".self_attn.q_norm" + ".weight" exists, a rms_norm will be built, if not it will go to else branch.
+std::shared_ptr<v1::Transpose> split_heads(const Output<Node>& x,
+                                            int num_h,
+                                            int  head_dim,
+                                            float rms_norm_eps,
+                                            const std::string& key,
+                                            const std::unordered_map<std::string, ov::Tensor>& consts) {
+    auto shape = std::make_shared<v0::Constant>(element::i64, Shape{4}, std::vector<int64_t>{0, 0, num_h, head_dim});
+    auto reshaped = std::make_shared<v1::Reshape>(x, shape, true);
+    if (consts.count(key + ".weight")) { //Qwen3 rms_norm
```

Comment:
Can we re-use make_rms_norm function below: https://github.com/openvinotoolkit/openvino.genai/blob/ea38c702a0657b459f9c9ba896ed8a6b2f25a838/src/cpp/src/gguf_utils/building_blocks.cpp#L763
If not, can you create a new function called `make_rms_norm_qwen3`, and check metadata architecture w/ "qwen3"?